tests/test_run_engine.py

```diff
@@ -139,3 +142,30 @@ def test_during_task_does_wait_with_small_timeout():
 
     event.wait.assert_called_with(0.1)
     assert event.wait.call_count == 2
+
+
+def test_run_engine_logs_all_documents(RE):
+    m = mock_open()
+    filepath: str
+    log_location = Path(
+        os.path.join("C:\\", "instrument", "var", "logs", "bluesky", "raw_documents")
+    )
+
+    def basic_plan() -> Generator[Msg, None, None]:
+        yield from bps.open_run()
+        yield from bps.close_run()
+
+    with patch("ibex_bluesky_core.callbacks.document_logger.open", m):
+        result: RunEngineResult = RE(basic_plan())
+        filepath = os.path.join(log_location, f"{result.run_start_uids[0]}.log")
+
+    for i in range(0, 3):
+        assert m.call_args_list[i].args == (filepath, "a")
```

Comment:
There's probably a nicer way to do this using `assert_calls` or something similar in the `unittest.mock` library?

Reply:
Tried to do this but got difficult